rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+
+## Non-goals
+
+1. Addressing anything related to tokens
+2. W3S corporate structure or nucleation plans
+
+## Background
+
+The current state of the world for web3.storage is as follows:
+
+- Web3.storage UCAN protocols are open and well specified. 
+
+- We've now created a version of these protocols that can be run from any machine-- so anyone can spin up theoretically own version of our service.
+
+- Seperately Web3.storage backs up data to Filecoin, via open protocols.
+
+- We are actively moving towards supporting running on top of underlying storage mechanisms that lack rich event functionality (a.k.a. bucket events)
+
+## Why Go Further
+
+Today, someone who is not the web3.storage team could spin up their own version of web3.storage, using the open source software we've published and the protocols we've specified. No one has done this in practice, but we are increasingly close to the point where it's plausable. Is this "sufficient decentralization"?
+
+One proposed path forward is to not pursue "decentralization" beyond trying to make other nodes a reality, perhaps through incentivization, and leaning heavily into product improvements that differentiate our platform. We could aim for "S3 if AWS invested in open source S3 implementations and made it content addressed".
+
+This is not the path we are going to pursue. It feels important to say that up front.
+
+But before we go deeper into where we are headed, it's worth looking at the implied dichotomy between "decentralization" and product features. That this dichotomy is almost reflexively referred to likely results from failing to appropriately frame "decentralization" in terms of product features. So let's start by framing the product features we want to unlock through decentralization.
+
+## The Network We Want
+
+### No Data Loss
+
+1. Users should be able to specify how many unique copies of their data they want backed up, and the W3S network should handle insuring that many copies remain backed up as long as the user pay to store in W3S.
+
+### Perpetuity
+
+1. Users should be able to store hot data in W3S that will be available for  retrieval until they stop paying for it. They should not have to worry about any single business entity holding their data disappearing. If it does, the data should go somewhere else without their needing to intervene.
+
+### Sensible defaults with customization
+
+1. Users should have the option to store with simply "the W3S network" rather than a specific business entity or Filecoin SP. This is a hard lesson learned from Filecoin that forcing users to choose from a marketplace where information on available options is hard to come by is a terrible user experience.
+
+2. Users should still have the option to choose a provider they like, or to move between providers.
```

Comment:
Do you mean filecoin storage provider here or something else ? I think interesting question in my head is do users make deals with w3s and w3s makes deals with SPs or do we want to allow users to make deals with SPs directly ? I recall requests for later on few occasions although I am not sure what the use cases were.

Reply:
I think users who just want to store with an SP directly should have that option.

I've come around to the fractal design concept -- an SP can run Filecoin.storage (or whatever we call Filecoin SP tailored w3up). If they do this, and they want to find their own users, they could just tell them to make direct data uploads to their instance of w3up, which would go in their own aggregation pipeline, and then eventually get put in their own deals for their Filecoin SP operation. and/or, they could connect their filecoin.storage instance to a web3.storage router, and get a portion of uploads going through us, which would likely just get aggregated along with any direct uploads they received.

Reply:
But seperately, you can also work directly with a provider running a hot storage instance with no Filecoin operation of their, who would then use a common aggregation pipeline (likely the one we run for a while).